games/applications/dj.py

```diff
@@ -0,0 +1,97 @@
+"""Deustch Jozsa Benchmark Circuits"""
+
+
+import os
+import numpy as np
+from qiskit import QuantumCircuit
+import pytest
+from . import backends, run_qiskit_circuit
+
+
+QASM_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "qasm")
+NUM = 5
+# -----------------------------------------------------------------------------------
+
+
+def constant(n):
+    """Creates a constant oracle"""
+    qc = QuantumCircuit(n + 1)
+    output = np.random.randint(2)
+    if output == 1:
+        qc.x(n)
+    c_oracle = qc.to_gate()
+    c_oracle.name = "Slayyy Oracle"
+    return c_oracle
+
+
+def balanced(n):
+    """Creates a balanced oracle"""
+    qc = QuantumCircuit(n + 1)
+    b = np.random.randint(1, 2**n)
+    bstr = format(b, "0" + str(n) + "b")
+    for i, qubit in enumerate(bstr):
+        if qubit == "1":
+            qc.x(i)
+    for qubit in range(n):
+        qc.cx(qubit, n)
+    for i, qubit in enumerate(bstr):
+        if qubit == "1":
+            qc.x(i)
+    b_oracle = qc.to_gate()
+    b_oracle.name = "Slayyy Oracle"
```

Comment:
Same here, while I like having fun names for things too since this ends up getting put in the qasm files we probably should use a more descriptive name for this gate. Probably something like `balanced_oracle`